### DISCUSSION TITLE
OAuthAccount in v10

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#993] [Q&A] [answered]
While updating to FastAPI-Users v10, I get this error in my application:
```
sqlalchemy.exc.IntegrityError: (sqlite3.IntegrityError) NOT NULL constraint failed: user.workspace
[SQL: INSERT INTO user (hashed_password, is_active, is_superuser, is_verified, id, anonymous, email, username, name, color, avatar, workspace, settings) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: ('$2b$12$Q7c7qea7Lw7.7Nx6PcvLleDgFc3X/I19hmqtq9rwis5irAKCGF1u2', 1, 0, 0, '7dfc5e0f-0a0f-4a6f-bb84-030b9b6efd0a', 1, 'david.brochart@gmail.com', None, None, None, None, None, None)]
```
I was previously overriding the `UserManager.on_after_register` method to [fill the mandatory fields and update the database](https://github.com/jupyter-server/jupyverse/blob/83b04598d51628a28fefaa2f3c1d00a09f38c758/plugins/auth/fps_auth/backends.py#L78-L97), but I'm a bit lost now and I'm not sure what the cause of the issue is.
It looks like the user is somehow automatically inserted into the database?

**Comment by frankie567** [comment, marked as answer]
> It looks like the user is somehow automatically inserted into the database?

It has always been like this: when you reach `on_after_register`, the user is already created in database. I think I know why you were not seeing the error before:

```py
class JupyterUser(BaseModel):
    anonymous: bool = True
    username: str = ""
    name: Optional[str] = None
    color: Optional[str] = None
    avatar: Optional[str] = None
    workspace: str = "{}"
    settings: str = "{}"
```
https://github.com/jupyter-server/jupyverse/blob/83b04598d51628a28fefaa2f3c1d00a09f38c758/plugins/auth/fps_auth/models.py#L7-L14

You were setting default values on the User and UserDB model. Since we were transforming the request data into UserDB before inserting, the DBMS was happy with those default values.

With the new version we directly try to build an SQLAlchemy `User` object from `UserCreate`... And there is no default values on those ones.

I can see two solutions then:

* Set default values for workspaces et al. on the SQLAlchemy `User` class. This would be the most transparent change IMO.
* Set those values as `nullable` in the DB, but there are probably unintended implications in the rest of your app.

**Reply by davidbrochart** [reply]
> Set default values for workspaces et al. on the SQLAlchemy User class

That works, thanks a lot François!
